iib/workers/tasks/build.py

```diff
@@ -833,20 +923,94 @@ def _adjust_operator_manifests(manifests_path):
         for pullspec in operator_csv.get_pullspecs():
             found_pullspecs.add(pullspec)
 
+    conf = get_worker_config()
+    registry_replacements = (
+        conf['iib_organization_customizations']
+        .get(organization, {})
+        .get('registry_replacements', {})
+    )
+
     # Resolve pull specs to container image digests
     replacement_pullspecs = {}
     for pullspec in found_pullspecs:
-        # Skip images that are already pinned
-        if ':' not in ImageName.parse(pullspec).tag:
-            replacement_pullspecs[pullspec] = ImageName.parse(_get_resolved_image(pullspec))
+        replacement_needed = False
+        if ':' in ImageName.parse(pullspec).tag:
+            resolved_image = ImageName.parse(pullspec)
+        else:
+            resolved_image = ImageName.parse(_get_resolved_image(pullspec))
+            replacement_needed = True
+
+        if registry_replacements.get(resolved_image.registry):
+            replacement_needed = True
+            log.debug(
```

Comment:
Could we move this log message to after the registry replacement operation and log the before and after pull spec? The method `ImageName.to_str()` should give you the full pull spec.